pkg/scan/trivy_scannable_test.go

```diff
@@ -0,0 +1,58 @@
+package scan
+
+import (
+	"testing"
+)
+
+func TestScannerType_String(t *testing.T) {
+	tests := []struct {
+		name     string
+		scanner  ScannerType
+		expected string
+	}{
+		{"SBOM Scanner", SBOMScannerType, "sbom"},
+		{"RootFS Scanner", RootFSScannerType, "rootfs"},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.scanner.String(); got != tt.expected {
+				t.Errorf("ScannerType.String() = %v, want %v", got, tt.expected)
+			}
+		})
+	}
+}
+
+func TestScannerType_Set(t *testing.T) {
+	tests := []struct {
+		name      string
+		input     string
+		expectErr bool
+	}{
+		{"Valid SBOM Scanner", "sbom", false},
+		{"Valid RootFS Scanner", "rootfs", false},
+		{"Invalid Scanner", "invalid", true},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var scanner ScannerType
+			err := scanner.Set(tt.input)
+			if (err != nil) != tt.expectErr {
```

Comment:
this is really clever, but would read better broken out as multiple conditions IMO

for example
```suggestion
			if tt.expectErr && err == nil {
```

I think it requires less cognitive overhead.

Reply:
Thanks for the suggestion!